### PR TITLE
fix(DB/Creature)Added paths for patrolling mobs in the Scarlet Abbey chapel area

### PR DESCRIPTION
**Pull request by fangshun2004**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. .go c 39959
2. go c 39998
3. go c 39967
4. go c 39978
5. go c 39980
6. go c 40015
7. go c 40019
8. go c 40008
9. go c 40034

无标题

## Known Issues and TODO List:

- [ ]Patrol mobs and some guid mobs randomly spawn different entry mobs. 
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by fangshun2004** [comment]
These paths are better with waypoint_data??

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+
+ -- Scarlet Wizard with guid 40034 smart ai
+SET @ENTRY := -40034;
+UPDATE `creature_template` SET `AIName` = 'SmartAI', `ScriptName` = '' WHERE `entry` = 4300;
+DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND `entryOrGuid` = @ENTRY;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+(@ENTRY, 0, 0, 0, 1, 0, 100, 0, 0, 0, 0, 0, 53, 0, 4003400, 1, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 'Once (OOC) - Self: Start path #4003400, walk, repeat, Aggressive');
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 22 AND `SourceEntry` = -40034 AND `SourceId` = 0;
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1713004478213792900.sql:268]: What does this refer to?

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+
+-- Replaces the generated coordinates
+DELETE FROM `creature` WHERE (`id1` = 4302) AND (`guid` IN (40031));
+INSERT INTO `creature` (`guid`, `id1`, `id2`, `id3`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES
+(40031, 4302, 0, 0, 189, 0, 0, 1, 1, 1, 1138.5015, 1369.3634, 30.3899, 1.4660766124725342, 86400, 2, 1, 3540, 2472, 1, 0, 0, 0, '', 0);
+
+DELETE FROM `creature_addon` WHERE (`guid` IN (40031));
+INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(40031, 0, 0, 0, 1, 0, 0, NULL);
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1713004478213792900.sql:292]: Need newline at the end

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+(@PATH, 14, 1045.2096 ,1448.4248 ,29.248976, NULL, 0, 'Scarlet Monastery Patrol');
+
+ -- Scarlet Myrmidon with guid 39967 smart ai
+SET @ENTRY := -39967;
+UPDATE `creature_template` SET `AIName` = 'SmartAI', `ScriptName` = '' WHERE `entry` = 4300;
+DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND `entryOrGuid` = @ENTRY;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+(@ENTRY, 0, 0, 0, 1, 0, 100, 0, 0, 0, 0, 0, 53, 0, 3996700, 1, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 'Once (OOC) - Self: Start path #3996700, walk, repeat, Aggressive');
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 22 AND `SourceEntry` = -39967 AND `SourceId` = 0;
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1713004478213792900.sql:71]: What did the conditions do?

**Comment by elthehablo** [comment]
> These paths are better with waypoint_data??

Waypoints is the correct table using smart scripts to trigger movement. So yes this is correct as it is now

**Comment by elthehablo** [comment]
Make title more descriptive/fit our PR standards please

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql (hunk 2 above)
Comment (fangshun2004) [reply data/sql/updates/pending_db_world/rev_1713004478213792900.sql:292]: conditions are automatically generated

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql (hunk 3 above)
Comment (fangshun2004) [reply data/sql/updates/pending_db_world/rev_1713004478213792900.sql:71]: conditions are automatically generated

File: data/sql/updates/pending_db_world/rev_1713004478213792900.sql (hunk 1 above)
Comment (fangshun2004) [reply data/sql/updates/pending_db_world/rev_1713004478213792900.sql:268]: I don't know why this thing exists. This is auto-generated. I can delete it if I don't need it

**Comment by elthehablo** [comment]
What did the conditions do?

**Comment by elthehablo** [comment]
I will check this again tonight

**Review by elthehablo — APPROVED** [review]

**Pull request merged**